src/OAuthAccessToken.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Creads\Partners;
+
+use GuzzleHttp\Client as GuzzleClient;
+
+class OAuthAccessToken implements AuthenticationInterface
+{
+    protected $baseUri = 'https://connect.creads-partners.com/';
+    protected $params = [];
+    protected $clientId;
+    protected $clientSecret;
+
+    public function __construct($clientId, $clientSecret, $params = [])
+    {
+        if (isset($params['base_uri'])) {
+            $this->baseUri = $params['base_uri'];
+        }
+
+        $this->clientId = $clientId;
+        $this->clientSecret = $clientSecret;
+        $this->params = $params;
+
+        if (isset($params['grant_type'])) {
+            if (in_array($params['grant_type'], ['client_credentials', 'password'])) {
+                throw new \Exception(sprintf('Unrecognized grant_type: %s', $params['grant_type']));
+            }
+            if ('password' === $params['grant_type'] && (!isset($params['username']) || !isset($params['password']))) {
+                throw new \Exception("'password' grant_type required a username and a password");
+            }
+        }
+    }
+
+    public function getConfig()
+    {
+        return [
+            'headers' => [
+                'Authorization' => 'Bearer '.$this->getAccessToken(),
+            ],
+        ];
+    }
+
+    public function getAccessToken($scope = 'base', $noCache = false)
```

Comment:
Actually, **cache** is not really the concept here. It's more a storing ability for the token than a cache system. So every cache references (variables or method) should be renamed. For instance :

`$noCache` would be `$noStoring`